lib/ledgerOperationStorage.js

```diff
@@ -35,6 +37,57 @@ module.exports = class LedgerOperationStorage {
       });
   }
 
+  // TODO: an optional parameter like
+  // `since: {blockHeight: 0, blockOrder: 0, eventOrder: 0}`
+  // could be useful if the state machine only needs operations after a certain
+  // point
+  getRecordHistory({recordId}, callback) {
+    assert.string(recordId, 'recordId');
+
+    const query = {recordId: database.hash(recordId)};
+
+    this.collection.aggregate([
+      {$match: query},
+      {$project: {_id: 0}},
+      {$lookup: {
+        from: this.eventCollectionName,
+        let: {eventHash: '$meta.eventHash'},
+        pipeline: [
+          {$match: {$expr: {$eq: ['$eventHash', '$$eventHash']}}},
+          {$project: {
+            _id: 0,
+            'meta.consensus': 1,
+            'meta.blockHeight': 1,
+            'meta.blockOrder': 1
+          }},
+          {$replaceRoot: {newRoot: "$meta"}}
+        ],
+        as: 'meta.eventMeta'
+      }},
+      {$unwind: '$meta.eventMeta'},
+      {$match: {'meta.eventMeta.consensus': {$exists: true}}},
+      {$sort: {
+        'meta.eventMeta.blockHeight': 1,
+        'meta.eventMeta.blockOrder': 1,
+        'meta.eventOrder': 1
+      }},
+    ], {allowDiskUse: true}).toArray((err, result) => {
+      if(err) {
+        return callback(err);
+      }
+      if(result.length === 0) {
+        return callback(new BedrockError(
+          'Failed to get history for the specified record.',
+          'NotFoundError', {
+            httpStatusCode: 404,
+            public: true,
+            recordId
+          }));
+      }
+      callback(null, result);
```

Comment:
I think this is fine for a first cut of this API. We may find that, for certain future use cases, there's "just too much history" to return it all at once. But we don't need to worry about that until we get closer to bumping into that problem. Just making a note here so it's in our heads.